CHANGELOG.md

```diff
@@ -1,3 +1,31 @@
+## 2.29.0-rc.0 / 2021-07-29
+
+* [CHANGE] Remove experimental tag on `--storage.tsdb.allow-overlapping-blocks` flag. #9117
+* [CHANGE] Remove experimental tag on `--storage.tsdb.retention.size` flag. #8728
+* [FEATURE] Add Kuma service discovery. #8844
+* [FEATURE] Add `__meta_gce_interface_ipv4_<name>` meta label to GCE discovery. #8978
+* [FEATURE] Add `__meta_ec2_availability_zone_id` meta label to EC2 discovery. #8978
+* [FEATURE] Add `__meta_hetzner_hcloud_labelpresent_<labelname>` meta label to Hetzner discovery. #9028
+* [FEATURE] Allow configuring exemplar storage via file and make it reloadable. #8974
+* [FEATURE] promtool: Add feature flags flag `--enable-feature`. #8958
+* [FEATURE] promtool: Add service discovery configuration validation. #8950
+* [ENHANCEMENT] Allow filtering with matchers in label-names API. #9083
+* [ENHANCEMENT] Add checkpoint replay to write-ahead-log replay status. #8898
+* [ENHANCEMENT] Log when total symbol size exceeds 2^32 bytes, causing compaction to fail. #9104
+* [ENHANCEMENT] Reduce blocking of outgoing remote write requests from series garbage collection. #9109
+* [ENHANCEMENT] Improve write-ahead-log decoding performance. #9106
+* [ENHANCEMENT] Improve write performance in TSDB. #9061
```

Comment:
```suggestion
* [ENHANCEMENT] Improve append performance in TSDB by reducing mutexes usage. #9061
```

TSDB is a lot, I would be more specific.

Reply:
fair enough